sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/pipeline_component.py

```diff
@@ -232,7 +231,13 @@ def _validate_binding_inputs(self, node: BaseNode) -> MutableValidationResult:
         not set. Raise error if pipeline input is optional but link to
         required inputs.
         """
-        component_definition_inputs = node.component.inputs
+        component_definition_inputs = {}
+        # Add flattened group input into definition inputs.
+        # e.g. {'group_name.item': PipelineInput}
+        for name, val in node.component.inputs.items():
+            if isinstance(val, GroupInput):
```

Comment:
Just an idea - do you think it's possible to use `isinstance(val, Input)` or even ensure that val is an Input instead of using isinstance(val, xxInput)?

Reply:
Sorry I don't get it, all of the val will be either Input or GroupInput, and for GroupInput we need to flatten them to insert some key like {'a.b': PipelineInput}, instead of {'a': GroupInput}

Reply:
I'm thinking about what if we use polymorphism instead of isinstance check. Theoretically, isinstance check should be kept only for direct user interface.

Reply:
Offline synced, will refine it somehow in the next pr